text/0510-dynamodb-global-table.md

```diff
@@ -0,0 +1,1742 @@
+scaling or with exact read capacity units. You would be paying for the provisioned capacity even if it is
+not utilized.
+
+The default mode is pay per request mode. If you choose provisioned mode, then you would need to specify read
+ and write capacity for the table and replicas. You can do this in three ways,
+
+1. Mention capacity on table level and these get copied over to all replicas.
+2. Mention capacity for each replica separately.
+3. Mention capacity on table level and for replica where you want to have a different capacity. Wherever you
+would not mention capacity, the table level values would be used.
+
+The billing mode will remain the same across all replicas.
+
+##### Pay Per Request Mode
+
+```typescript
+  new GlobalTable(stack, 'FooTable', {
+    partitionKey: { name: 'Foo', type: AttributeType.STRING },
+    billingMode: BillingMode.PAY_PER_REQUEST,
+  });
+```
+
+##### Provisioned Mode
+
+* __Table level configuration with auto scaling__
+
+```typescript
+  new GlobalTable(stack, 'FooTable', {
+    partitionKey: { name: 'Foo', type: AttributeType.STRING },
+    billingMode: BillingMode.PROVISIONED,
+    writeWithAutoScaling: {
+      max: 50,
+    },
+    readWithAutoScaling: {
+      max: 50,
+    },
+  });
+```
+
+* __Table level configuration read capacity without auto scaling__
+
+```typescript
+  new GlobalTable(stack, 'FooTable', {
+    partitionKey: { name: 'Foo', type: AttributeType.STRING },
+    billingMode: BillingMode.PROVISIONED,
+    writeWithAutoScaling: {
+      max: 50,
+    },
+    readCapactity: 50,
+  });
+```
+
+__NOTE:__
+
+* Provisioned mode for write capacity could only be used with autoscaling. There is no way to provision
+an exact write capacity unit in global tables.
+* Write capacity could not be configured on a per replica basis. You can only modify read capacity per
+replica.
+* You can either define `readCapacity` or `readWithAutoScaling` but not both at the same time.
+
+#### Global Secondary Indexes
+
+You can add global secondary indexes to your global table. These would be the same for each replica of the
+table. There is some configuration that could be specified per replica.
+
+```typescript
+  const globalTable = new GlobalTable(stack, 'FooTable', {
+    partitionKey: { name: 'Foo', type: AttributeType.STRING },
+    sortKey: { name: 'Bar', type: AttributeType.STRING },
+    globalSecondaryIndex: [
+      {
+        indexName: 'FooTableGsi',
+        partitionKey: { name: 'Bar', type: AttributeType.STRING },
+      },
+    ],
+  });
+```
+
+You can add global secondary index with `addGlobalSecondaryIndex` method.
+
+```typescript
+  globalTable.addGlobalSecondaryIndex({
+    indexName: 'FooTableGsi',
+    partitionKey: { name: 'Bar', type: AttributeType.STRING },
+  });
+```
+
+__NOTE:__
+
+* You can create up to 20 global secondary indexes.
+* You can only create or delete one global secondary index in a single stack operation.
```

Comment:
WTF!!!! 

Is this true? 

When I have a table with 5 GSIs, and I want to deploy it to a new region (or create a new replica), is that not possible?

Because if so, we need to make provisions for that so that your definition can be the same but we temporarily turn off GSIs or something so CloudFormation doesn't attempt to deploy them.